### PR TITLE
NavBar links and routes implemented

### PR DESCRIPTION
**Pull request by aleksaalfir**
Nav bar links are now displaying based on user role. User now gets redirected if he tries to visit unauthorised route.

### CODE REVIEW

**Review by milos-majkic — APPROVED** [review]

File: src/pages/BooksPage/BooksPage.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import Button from 'components/core/Button/Button';
+import { getBooks } from 'services/api/bookService';
+
+const BooksPage: React.FC = () => {
+  const [books, setBooks] = useState<Book[]>([]);
+  const [searchInput, setSearchInput] = useState<string>('');
+
+  const searchBooksHandler = (): void => {
+    getBooks().then((data) => {
+      const dataBooks: Book[] = data.content;
```
Comment (petarnqode) [inline src/pages/BooksPage/BooksPage.tsx]: Can we do this in other way, const { content : dataBooks } = data; ?

File: src/pages/BooksPage/BooksPage.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        {books.map((book, index) => {
+          return (
+            <BookCard
+              key={index}
+              id={book.id}
+              title={book.title}
+              author={book.author}
+              description={book.description}
+              imagePath={book.imagePath}
+              numOfCopies={book.numOfCopies}
```
Comment (petarnqode) [inline src/pages/BooksPage/BooksPage.tsx]: This can get messy, can we send book obj and destruct in component or {...book} ?

**Review by petarnqode — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,28 +1,24 @@
...
-        <Route path="/rent/:id" element={<RentedBookPage />} />
+        {getAllowedRoutes().map((route, index) => {
+          return (
+            <Route
+              path={route.path}
+              element={
+                route.layout ? <StandardLayout>{route.element}</StandardLayout> : route.element
+              }
+              key={index}
+            />
```
Comment (vkotarlic) [inline src/App.tsx]: I would like to handle this in different way

File: src/components/Layout/Header/Header.tsx [hunk 4]
```diff
@@ -7,11 +10,18 @@ const Header = () => {
...
-          <li className={classes['c-header__nav-link']}>Sales</li>
-          <li className={classes['c-header__nav-link']}>My books</li>
-          <li className={classes['c-header__nav-link']}>Profile</li>
+          {getAllowedNavbarLinks().map((link, index) => {
+            return (
+              <NavLink to={link.url} key={index} className={classes['c-header__nav-link']}>
+                {link.text}
+              </NavLink>
+            );
+          })}
```
Comment (vkotarlic) [inline src/components/Layout/Header/Header.tsx]: It is same for this part, you are mapping list of links but i don't think it is good idea

File: src/pages/BooksPage/BooksPage.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  const searchBooksHandler = (): void => {
+    getBooks().then((data) => {
+      const { content: dataBooks } = data;
+
+      setBooks(
+        dataBooks.filter(
+          (book: Book) =>
+            book.title.toLocaleLowerCase().includes(searchInput.toLocaleLowerCase()) ||
+            book.author.toLocaleLowerCase().includes(searchInput.toLocaleLowerCase())
+        )
```
Comment (vkotarlic) [inline src/pages/BooksPage/BooksPage.tsx]: I would like to filter books outside of setState and then just set them

**Review by vkotarlic — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (aleksaalfir) [reply src/App.tsx]: Can i have an additional explanation?

**Pull request merged**